### ISSUE TITLE
UI Overhaul

### ISSUE THREAD

**Issue by prysmanta** [godotengine/godot-proposals#2849]
### Describe the project you are working on
A Godot UI Overhaul, and a rebrand.

### Describe the problem or limitation you are having in your project
The current Godot interface looks like a toy for an incoming user. This may present problems for newbies or people who do not know how to code and program, aswell as the professionals seeking another engine.

### Describe the feature / enhancement and how it helps to overcome the problem or limitation
The new UI overhaul presents 3 improvements: 

- A professional looking interface fit for the general developer
- Monetization of Godot through high quality and unintrusive advertisements
- An easier and cleaner workflow for both new and old developers

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams
![Godot NextArtboard 1 copy](https://user-images.githubusercontent.com/65317109/121546125-a54c7800-ca3d-11eb-89a2-27443700b50f.png)
This is the new presented launcher. It is not a seperate executable, meaning Godot can still be in one compact and portable executable. The sidebar presents 6 options: 

- Home: Where the largest banner presenting sponsored content, new games, and new godot updates are present. It also has a button for creating a new project, uploading assets, or sending feedbacks and issues. Not only that, it also has 3 sub categories which are news, events, and pinned news & events. These 3 serve as web portals for the latest news and development related advertisements which all open in the browser. The pinned submenu serves as an archive for said content.
- Projects: The general list of user-made projects. The user can go to settings to launch godot automatically on this menu instead of home.
- Assets: The Asset manager. Individually manage downloaded assets and download more through a large button on the top-right.
- Editors: Download several versions of Godot through here. See if a version is brand new, and delete the current version you have right here aswell. (A new instance of godot will launch with the newer version, closing the older one)
- Account: User credentials are stored right here. Only a username and email is needed, to automatically fill out anything that needs the credentials. (A local account of sorts. Can be disregarded upon setup)
- Settings: A quick shortcut to the settings panel, to adjust things before opening the editor itself.

//ps there shouldve been icons for the new project, upload asset, and feedback buttons but i was too burnt out.

![Godot NextArtboard 1 copy 2](https://user-images.githubusercontent.com/65317109/121548692-ce6e0800-ca3f-11eb-8ccd-8930066925ca.png)
This is the splash screen. It blurs out the launcher to present a loading screen when opening a project. The editor should have an animation that fades-in from the bottom when a project successfully opens.

![Godot NextArtboard 1](https://user-images.githubusercontent.com/65317109/121549087-20169280-ca40-11eb-91a7-b63b934d4795.png)
This is the editor itself. Following the footsteps of unreal, the filesystem is hidden in a dock to save screenspace. Renamed as "content archive" to manage all the imported objects right where it needs to be. The panels still function as they are, albeit with just a minor redesign

### If this enhancement will not be used often, can it be worked around with a few lines of script?
This is a large enhancement, so a few lines or even a large block of script cannot work these around.

### Is there a reason why this should be core and not an add-on in the asset library?
It's a whole overhaul. A possible improvement for all user experiences.

**Comment by Calinou** [comment]
The editor theme was revamped by yours truly for 4.0: https://github.com/godotengine/godot/pull/45607

**Comment by KoBeWi** [comment]
Project manager with news section and advertisements is an overkill. Although it's perfectly possible to make one as a 3rd party app and use instead of the default manager (and it doesn't even require any changes to source code).

**Comment by Calinou** [comment]
> Project manager with news section and advertisements is an overkill. Although it's perfectly possible to make one as a 3rd party app and use instead of the default manager (and it doesn't even require any changes to source code).

See [Hourglass](https://hourglass.flyingpimonster.net/) for an example of this :slightly_smiling_face:

**Comment by SilencedPerson** [comment]
Honestly, i don't like the proposal as a whole but there are some nice ideas you sneaked into your mockup picture.
1) less visual clutter, in particular, less borders all around and less rectangles in the inspector.
2) the subsettings in the settings menu being moved to separate column makes them much more readable at a first glance.
3) removal of the file icon from the file system, makes it much easier to read.

I did a mockup based on the current version of the GUI made by @Calinou and on the suggestion to remove the inner borders by someone on twitter. (I believe it was JFons but can't find it at the moment.)
![image](https://user-images.githubusercontent.com/82024211/121694243-3c5f1180-caca-11eb-938e-263c7fbb9047.png)

**Comment by pycbouh** [comment]
> on the suggestion to remove the inner borders by someone on twitter. (I believe it was JFons but can't find it at the moment.)

Here's JFons' original mockup from https://chat.godotengine.org/channel/editor?msg=toFzFh9oAYBHSbnAa

![mockup](https://user-images.githubusercontent.com/11782833/121696537-d6c45280-cad4-11eb-8eea-a0e18d52b24b.png)

**Comment by me2beats** [comment]
In the whole I find the theme kinda cleaner.
I like main colors, rounded corners, especially for text fields, the loader, the controls design.
I don't like the proposed Godot icon design (looks scary imo),
Suggested icons are designed in a bit different style, I like the default ones, they look consistent and readable.

**Comment by bmourit** [comment]
Well, I agree. The Editor needs a major overhaul if it wants to be seen as professional. See what Unity and Unreal have done. By comparison this comes across as cheap and almost cartoonish. Further, nothing is easy to find. So example materials. It is nested under several menus with no clarity on where it should actually go. These types of things need to be easy to find, not hidden under various obscure levels of hierarchy. Coming from another Engine, this was something I found too burdensome for iterating. 
Everything seems to be hidden in hierarchies, without any clear explanation as to why, which makes it extremely annoying to try and find something. It should, in contrast, be intuitive.

**Comment by tkgalk** [comment]
I agree, the Editor definitely needs work and a good UX/UI design pass. It's not looking... too professional. I can agree that it looks like it belongs in school at your "Game Development" classes. Not that it's a bad thing, but it doesn't exude confidence as a serious tool.

I think when it comes to design language current Blender UI components and style do a good job at that.

**Comment by bmourit** [comment]
I agree about Blender being a good example of what an improvement might look like.
